src/controllers/wallet/tx-proposal/tx-proposal.controller.js

```diff
@@ -0,0 +1,247 @@
+    res.send({ success: true, txHex: tx.toHex(), dataToSignHash: tx.getDataToSignHash().toString('hex') });
+  } catch (err) {
+    res.send({ success: false, error: err.message });
+  } finally {
+    lock.unlock(lockTypes.SEND_TX);
+  }
+}
+
+async function addSignatures(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const { txHex, signatures } = req.body;
+  const tx = helpersUtils.createTxFromHex(txHex, req.wallet.getNetworkObject());
+
+  for (const signature of signatures) {
+    tx.inputs[signature.index].setData(Buffer.from(signature.data, 'hex'));
+  }
+
+  res.send({ success: true, txHex: tx.toHex() });
+}
+
+async function pushTxHex(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const canStart = lock.lock(lockTypes.SEND_TX);
+  if (!canStart) {
+    res.send({ success: false, error: cantSendTxErrorMessage });
+    return;
+  }
+
+  const network = req.wallet.getNetworkObject();
+  const { txHex } = req.body;
+
+  try {
+    const tx = helpersUtils.createTxFromHex(txHex, req.wallet.getNetworkObject());
+    const sendTransaction = new SendTransaction({ transaction: tx, network });
+    const response = await sendTransaction.runFromMining();
+    res.send({ success: true, tx: mapTxReturn(response) });
+  } catch (err) {
+    res.send({ success: false, error: err.message });
+  } finally {
+    lock.unlock(lockTypes.SEND_TX);
+  }
+}
+
+/**
+ * Get metadata on all inputs from the loaded wallet.
+ */
+function getWalletInputs(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const { txHex } = req.query;
+  const { wallet } = req;
+
+  try {
+    const network = wallet.getNetworkObject();
+    const tx = helpersUtils.createTxFromHex(txHex, network);
+    res.send({ success: true, inputs: wallet.getWalletInputInfo(tx) });
+  } catch (err) {
+    res.send({ success: false, error: err.message });
+  }
+}
+
+function getInputData(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const { index, type } = req.body;
+  storage.setStore(req.wallet.store);
+  const accessData = storage.getItem('wallet:accessData');
+  let inputData;
+
+  if (type === 'p2pkh') {
+    const { signature } = req.body.signature;
+    inputData = transactionUtils.createInputData(
+      Buffer.from(signature, 'hex'),
+      walletUtils.getPublicKeyFromXpub(accessData.xpubkey, index).toBuffer(),
+    );
+  } else if (type === 'p2sh') {
+    // check if the loaded wallet is MultiSig
+    const multisigData = accessData.multisig;
+    if (multisigData) {
```

Comment:
Shouldn't this be `!multisigData`?

Reply:
How was it working with the tests like this? Did you add a new test for this to fail?

Reply:
When running locally I made a filter to run the tx-proposal tests, but I think when I filtered out the multisig tests it also filtered out the multisig cases of the tx-proposal tests. If you check the fixes when the CI was failing most of them were on p2sh cases.